### DISCUSSION TITLE
Avoiding errors on as-yet-undefined syntax

### DISCUSSION THREAD

**Discussion by TonyGravagno** [nene/sql-parser-cst#91] [Ideas]
I was going to post this suggestion here but I brain dumped in [this issue](https://github.com/nene/sql-parser-cst/issues/89#issuecomment-2438592605).

Might this be a way to allow the plugin to process whatever it can, and ignore whatever it doesn't recognize?

After the joy of finding this plugin (and the enhancements over prior tooling) I was saddened that any unrecognized syntax would cause a termination of the parsing process, resulting in a loss of whatever **could** be processed.

**Comment by nene** [comment]
Yeah, I think this sort of approach should work and should be pretty simple to implement.

It should be enabled with some flag though, so it could be toggled on/off.

**Comment by nene** [comment]
I've now finally implemented this suggestion. One can now use the `acceptUnsupportedGrammar` option (or `sqlAcceptUnsupportedGrammar` in the prettier plugin) to enable it.
